pysph/tools/interpolator.py

```diff
@@ -209,19 +313,30 @@ def interpolate(self, prop, gradient=False):
         prop: str
             The name of the property to interpolate.
 
-        gradient: bool
-            Evaluate gradient and not function.
+        comp: int
+            The component of the gradient required
 
         Returns
         -------
         A numpy array suitably shaped with the property interpolated.
         """
+        assert isinstance(comp, int), 'Error: only interger value is allowed'
         for array in self.particle_arrays:
             data = array.get(prop, only_real_particles=False)
             array.get('temp_prop', only_real_particles=False)[:] = data
 
         self.func_eval.compute(0.0, 0.1)  # These are junk arguments.
-        result = self.pa.prop.copy()
+        if comp and self.use_shepard:
+            print("Error: set use_shepard=False to evaluate gradients")
+            result = np.zeros_like(self.pa.x)
+        elif self.use_shepard:
```

Comment:
I think it makes sense to change the `use_shepard` option to an `order` or `method` option which has three possibilities ('sph', 'shepard', 'order1') and you can add an assert statement to ensure this in the `__init__`, you can keep `'shepard'` as the default.

Reply:
done